umbraco/Umbraco-CMS
# 14744 tour not coded to best practice

### Prerequisites


If there's an existing issue for this PR then this fixes [14744](https://github.com/umbraco/Umbraco-CMS/issues/14744)

### Description

The following changes were made:

1. The focusLockService function is now passed into the TourStepDirective function in umbtourstep.directive.js.
2. HTML properties for accessibility were added to umb-tour.html, umb-tour-step-content.html, umb-tour-step-counter.html, umb-tour-step-footer.html, umb-tour-step-header.html. These properties include tabindex, role, and aria-label.

Prerequisites to test:

1. A screen reader has been downloaded on the PC being used to test. NVDA was used to test during development.

Steps to test:

1. From the Umbraco home page, open any tour via the right side panel.
2. With the screen reader enabled, press TAB until the dialog is reached. "Tour Dialog" should be announced when the dialog is reached.
3. Continue to press TAB. For each element, the content of the element will be announced by the screen reader.
4. When the last button is reached, press Shift + Tab to navigate backwards through the dialog. All contents should be announced by the screen reader.

Note that for certain screen readers and certain screen reader versions, the behavior may be unexpected due to a bug in the screen reader itself. Certain browser versions may also have bugs that produce unexpected behaviors with screen readers.

No unit tests were added, though I can add a few if necessary.
